geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterIngest.scala

```diff
@@ -0,0 +1,107 @@
+      rsp.writeMemoryParam.getName  -> config(IngestRasterParams.WRITE_MEMORY),
+      dsp.writeThreadsParam         -> config(IngestRasterParams.WRITE_THREADS),
+      dsp.queryThreadsParam.getName -> config(IngestRasterParams.QUERY_THREADS),
+      dsp.mockParam.getName         -> config(IngestRasterParams.ACCUMULO_MOCK)
+    ).collect {
+      case (key, Some(value)) => (key, value);
+      case (key, value: String) => (key, value)
+    }).asInstanceOf[java.util.Map[String, Serializable]]
+
+  def createRasterStore(config: Map[String, Option[String]]): AccumuloRasterStore = {
+    val rasterName = config(IngestRasterParams.TABLE)
+    if (rasterName == null || rasterName.isEmpty) {
+      logger.error("No raster name specified for raster feature ingest." +
+        " Please check that all arguments are correct in the previous command. ")
+      sys.exit()
+    }
+
+    val csConfig: JMap[String, Serializable] = getAccumuloRasterStoreConf(config)
+
+    AccumuloRasterStore(csConfig)
+  }
+
+  def createRasterStore(args: Args): AccumuloRasterStore = {
+    val dsConfig: Map[String, Option[String]] =
+      Map(
+        IngestRasterParams.ZOOKEEPERS        -> args.optional(IngestRasterParams.ZOOKEEPERS),
+        IngestRasterParams.ACCUMULO_INSTANCE -> args.optional(IngestRasterParams.ACCUMULO_INSTANCE),
+        IngestRasterParams.ACCUMULO_USER     -> args.optional(IngestRasterParams.ACCUMULO_USER),
+        IngestRasterParams.ACCUMULO_PASSWORD -> args.optional(IngestRasterParams.ACCUMULO_PASSWORD),
+        IngestRasterParams.TABLE             -> args.optional(IngestRasterParams.TABLE),
+        IngestRasterParams.GEOSERVER_REG     -> args.optional(IngestRasterParams.GEOSERVER_REG),
+        IngestRasterParams.AUTHORIZATIONS    -> args.optional(IngestRasterParams.AUTHORIZATIONS),
+        IngestRasterParams.VISIBILITIES      -> args.optional(IngestRasterParams.VISIBILITIES),
+        IngestRasterParams.WRITE_MEMORY      -> args.optional(IngestRasterParams.WRITE_MEMORY),
+        IngestRasterParams.WRITE_THREADS     -> args.optional(IngestRasterParams.WRITE_THREADS),
+        IngestRasterParams.QUERY_THREADS     -> args.optional(IngestRasterParams.QUERY_THREADS),
+        IngestRasterParams.ACCUMULO_MOCK     -> args.optional(IngestRasterParams.ACCUMULO_MOCK)
+      )
+    createRasterStore(dsConfig)
+  }
+
+  def getReader(imageFile: File, imageType: String): AbstractGridCoverage2DReader = {
+    imageType match {
+      case TIFF => getTiffReader(imageFile)
+      case DTED => getDtedReader(imageFile)
+      case _ => throw new Exception("Image type is not supported.")
+    }
+  }
+
+  def getTiffReader(imageFile: File): AbstractGridCoverage2DReader = {
+    new GeoTiffReader(imageFile, new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, true))
+  }
+
+  def getDtedReader(imageFile: File): AbstractGridCoverage2DReader = {
+    val l = new ImageLayout()
+    l.setTileGridXOffset(0).setTileGridYOffset(0).setTileHeight(512).setTileWidth(512)
```

Comment:
This is just a property set for JAI and can actually be dropped entirely I believe.